File: 252_Flexible/252_Flexible.js
function solution(schedules, timelogs, startday) {
  let answer = 0;

  timelogs.map((timelog, idx) => {
    let flag = true,
      day = startday - 1;

    for (const table of timelog) {
      day++;
      if (day === 6 || day === 7) continue;

      const tHour = Math.floor(Number(table) / 100),
        tMinute = Number(table) % 100;
      const sHour = Math.floor(Number(schedules[idx]) / 100),
        sMinute = Number(schedules[idx] % 100);
      const diff = (tHour - sHour) * 60 + (tMinute - sMinute);
      if (diff > 10) flag = false;

      if (day > 7) day = 1;
    }
    if (flag) answer++;
  });
  return answer;
}

File: 252_Flexible/252_Flexible.py
def solution(schedules, timelogs, startday):
    answer = 0
    
    def convertTime(time):
        h = time // 100
        m = time % 100
        return h * 60 + m
    
    for idx, timelog in enumerate(timelogs):
        s = startday - 1
        flag = True
        for table in timelog:
            s += 1
            if s in [6, 7]:
                continue
            elif s > 7:
                s = 1
                
            if convertTime(table) - convertTime(schedules[idx]) > 10:
                flag = False
        
        if flag:
            answer += 1
            
        
    
    return answer